### PR TITLE
Added code to handle votes in frontend and backend

### PR DESCRIPTION
**Pull request by palvitgarg99**

### CODE REVIEW

File: src/Components/ProductTile.js [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
```
Comment (elric97) [inline src/Components/ProductTile.js:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogTitle from "@mui/material/DialogTitle";
```

File: src/Components/ProductTile.js [hunk 2]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleDownVote = () =>{
+    const form = new FormData();
+    form.append("emailId", username);
+    form.append("productId", index);
+    Service.post('removeVote', form)
```
Comment (elric97) [inline src/Components/ProductTile.js:102]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("removeVote", form)
```

File: src/Components/ProductTile.js [hunk 3]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+  const handleClose = () => {
+    setOpen(false);
   };
+
+  const increaseUpVote = () =>{
+    const form = new FormData();
+    console.log(products[index].uid);
+    form.append("uid", products[index].uid);
+    Service.post('addTotalVote', form)
+      .then(data =>
```
Comment (elric97) [inline src/Components/ProductTile.js:42]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
      .then((data) => 
```

File: src/Components/ProductTile.js [hunk 4]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
-    const updatedProduct = { ...products[index] };
-    let currentVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.upVoted = !updatedProduct.upVoted;
-    updatedProduct.downVoted = false;
-    let newVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.votes = updatedProduct.votes - currentVote + newVote;
-    setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+
+  const username = ReactSession.get("username");
+  const [loggedin, setLoggedin] = useState(username !== '');
```
Comment (elric97) [inline src/Components/ProductTile.js:26]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const [loggedin, setLoggedin] = useState(username !== "");
```

File: src/Components/ProductTile.js [hunk 5]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          }
+          console.log(data.code);
+          if (data.code > 200) {
+            console.log("Error")
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleDownVote = () =>{
```
Comment (elric97) [inline src/Components/ProductTile.js:98]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const handleDownVote = () => {
```

File: src/Components/ProductTile.js [hunk 6]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleUpVote = () =>{
+    const form = new FormData();
+    form.append("emailId", username);
+    form.append("productId", index);
+    Service.post('addVote', form)
```
Comment (elric97) [inline src/Components/ProductTile.js:62]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("addVote", form)
```

File: src/Components/ProductTile.js [hunk 7]
```diff
@@ -1,6 +1,15 @@
+import React, { useState } from 'react';
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
```
Comment (elric97) [inline src/Components/ProductTile.js:10]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogContentText from "@mui/material/DialogContentText";
```

File: src/Components/ProductTile.js [hunk 8]
```diff
@@ -1,6 +1,15 @@
+import React, { useState } from 'react';
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
```
Comment (elric97) [inline src/Components/ProductTile.js:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Dialog from "@mui/material/Dialog";
```

File: src/Components/ProductTile.js [hunk 9]
```diff
@@ -1,6 +1,15 @@
+import React, { useState } from 'react';
```
Comment (elric97) [inline src/Components/ProductTile.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import React, { useState } from "react";
```

File: src/Components/ProductTile.js [hunk 10]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+
+  const decreaseUpVote = () =>{
+    const form = new FormData();
+    form.append("uid", products[index].uid);
+    Service.post('subTotalVote', form)
```
Comment (elric97) [inline src/Components/ProductTile.js:81]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("subTotalVote", form)
```

File: src/Components/ProductTile.js [hunk 11]
```diff
@@ -1,6 +1,15 @@
+import React, { useState } from 'react';
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
```
Comment (elric97) [inline src/Components/ProductTile.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogActions from "@mui/material/DialogActions";
```

File: src/Components/ProductTile.js [hunk 12]
```diff
@@ -1,6 +1,15 @@
...
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import Service from '../Service';
```
Comment (elric97) [inline src/Components/ProductTile.js:12]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Service from "../Service";
```

File: src/Components/ProductTile.js [hunk 13]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+
+  const handleClose = () => {
+    setOpen(false);
   };
+
+  const increaseUpVote = () =>{
+    const form = new FormData();
+    console.log(products[index].uid);
+    form.append("uid", products[index].uid);
+    Service.post('addTotalVote', form)
```
Comment (elric97) [inline src/Components/ProductTile.js:41]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("addTotalVote", form)
```

File: src/Components/ProductTile.js [hunk 14]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          }
+          console.log(data.code);
+          if (data.code > 200) {
+            console.log("Error")
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleDownVote = () =>{
```
Comment (elric97) [inline src/Components/ProductTile.js:98]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const handleDownVote = () => {
```

File: src/Components/ProductTile.js [hunk 15]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
-    updatedProduct.upVoted = false;
-    let newVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.votes = updatedProduct.votes - currentVote + newVote;
-    setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+
+  const handleClose = () => {
+    setOpen(false);
   };
+
+  const increaseUpVote = () =>{
```
Comment (elric97) [inline src/Components/ProductTile.js:37]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const increaseUpVote = () => {
```

File: src/Components/ProductTile.js [hunk 16]
```diff
@@ -1,6 +1,15 @@
+import React from 'react';
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
```
Comment (elric97) [inline src/Components/ProductTile.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Button from "@mui/material/Button";
```

File: src/Components/ProductTile.js [hunk 17]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleUpVote = () =>{
+    const form = new FormData();
+    form.append("emailId", username);
+    form.append("productId", index);
+    Service.post('addVote', form)
```
Comment (elric97) [inline src/Components/ProductTile.js:62]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("addVote", form)
```

File: src/Components/ProductTile.js [hunk 18]
```diff
@@ -1,6 +1,15 @@
+import React from 'react';
```
Comment (elric97) [inline src/Components/ProductTile.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import React from "react";
```

File: src/Components/ProductTile.js [hunk 19]
```diff
@@ -1,6 +1,15 @@
...
+import { ReactSession } from 'react-client-session';
```
Comment (elric97) [inline src/Components/ProductTile.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import { ReactSession } from "react-client-session";
```

File: src/Components/ProductTile.js [hunk 20]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          }
+          if (data.code > 200) {
+            console.log("Error")
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+
+  const decreaseUpVote = () =>{
```
Comment (elric97) [inline src/Components/ProductTile.js:78]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const decreaseUpVote = () => {
```

File: src/Components/ProductTile.js [hunk 21]
```diff
@@ -1,6 +1,15 @@
+import React from 'react';
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
```
Comment (elric97) [inline src/Components/ProductTile.js:9]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogContent from "@mui/material/DialogContent";
```

File: src/Components/ProductTile.js [hunk 22]
```diff
@@ -1,6 +1,15 @@
+import React from 'react';
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
```
Comment (elric97) [inline src/Components/ProductTile.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogActions from "@mui/material/DialogActions";
```

File: src/Components/ProductTile.js [hunk 23]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+
+  const decreaseUpVote = () =>{
+    const form = new FormData();
+    form.append("uid", products[index].uid);
+    Service.post('subTotalVote', form)
```
Comment (elric97) [inline src/Components/ProductTile.js:81]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("subTotalVote", form)
```

File: src/Components/ProductTile.js [hunk 24]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          }
+          console.log(data.code);
+          if (data.code > 200) {
+            console.log("Error")
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleUpVote = () =>{
```
Comment (elric97) [inline src/Components/ProductTile.js:58]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const handleUpVote = () => {
```

File: src/Components/ProductTile.js [hunk 25]
```diff
@@ -1,6 +1,15 @@
+import React from 'react';
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
```
Comment (elric97) [inline src/Components/ProductTile.js:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Dialog from "@mui/material/Dialog";
```

File: src/Components/ProductTile.js [hunk 26]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+
+  const handleClose = () => {
+    setOpen(false);
   };
+
+  const increaseUpVote = () =>{
+    const form = new FormData();
+    console.log(products[index].uid);
+    form.append("uid", products[index].uid);
+    Service.post('addTotalVote', form)
```
Comment (elric97) [inline src/Components/ProductTile.js:41]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("addTotalVote", form)
```

File: src/Components/ProductTile.js [hunk 27]
```diff
@@ -1,6 +1,15 @@
...
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import Service from '../Service';
```
Comment (elric97) [inline src/Components/ProductTile.js:12]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Service from "../Service";
```

File: src/Components/ProductTile.js [hunk 28]
```diff
@@ -1,6 +1,15 @@
...
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
```
Comment (elric97) [inline src/Components/ProductTile.js:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogTitle from "@mui/material/DialogTitle";
```

File: src/Components/ProductTile.js [hunk 29]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+          } else {
+            console.log(data.success)
+          }
+        });
+  }
+  const handleDownVote = () =>{
+    const form = new FormData();
+    form.append("emailId", username);
+    form.append("productId", index);
+    Service.post('removeVote', form)
```
Comment (elric97) [inline src/Components/ProductTile.js:102]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("removeVote", form)
```

File: src/Components/ProductTile.js [hunk 30]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
-    const updatedProduct = { ...products[index] };
-    let currentVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.upVoted = !updatedProduct.upVoted;
-    updatedProduct.downVoted = false;
-    let newVote = updatedProduct.upVoted ? 1 : (updatedProduct.downVoted ? -1 : 0);
-    updatedProduct.votes = updatedProduct.votes - currentVote + newVote;
-    setProducts(products.map((product) => product.id === products[index].id ? updatedProduct : product));
+
+  const username = ReactSession.get("username");
+  const loggedin = username !== ''?true:false;
```
Comment (elric97) [inline src/Components/ProductTile.js:26]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const loggedin = username !== ""?true:false;
```

File: src/Components/ProductTile.js (hunk 16 above)
Comment (elric97) [inline src/Components/ProductTile.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Button from "@mui/material/Button";
```

File: src/Components/ProductTile.js (hunk 17 above)
Comment (elric97) [inline src/Components/ProductTile.js:62]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("addVote", form)
```

File: src/Components/ProductTile.js (hunk 26 above)
Comment (elric97) [inline src/Components/ProductTile.js:41]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("addTotalVote", form)
```

File: src/Components/ProductTile.js (hunk 25 above)
Comment (elric97) [inline src/Components/ProductTile.js:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Dialog from "@mui/material/Dialog";
```

File: src/Components/ProductTile.js [hunk 31]
```diff
@@ -12,24 +21,100 @@ import { useHistory } from 'react-router';
...
+  const handleClose = () => {
+    setOpen(false);
   };
+
+  const increaseUpVote = () =>{
+    const form = new FormData();
+    console.log(products[index].uid);
+    form.append("uid", products[index].uid);
+    Service.post('addTotalVote', form)
+      .then(data =>
```
Comment (elric97) [inline src/Components/ProductTile.js:42]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
      .then((data) => 
```

File: src/Components/ProductTile.js (hunk 14 above)
Comment (elric97) [inline src/Components/ProductTile.js:98]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const handleDownVote = () => {
```

File: src/Components/ProductTile.js (hunk 24 above)
Comment (elric97) [inline src/Components/ProductTile.js:58]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const handleUpVote = () => {
```

File: src/Components/ProductTile.js [hunk 32]
```diff
@@ -1,6 +1,15 @@
+import React from 'react';
+import { ReactSession } from 'react-client-session';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faChevronDown, faChevronUp } from '@fortawesome/free-solid-svg-icons';
 import { useHistory } from 'react-router';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
```
Comment (elric97) [inline src/Components/ProductTile.js:10]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogContentText from "@mui/material/DialogContentText";
```

File: src/Components/ProductTile.js (hunk 22 above)
Comment (elric97) [inline src/Components/ProductTile.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogActions from "@mui/material/DialogActions";
```

File: src/Components/ProductTile.js (hunk 23 above)
Comment (elric97) [inline src/Components/ProductTile.js:81]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("subTotalVote", form)
```

File: src/Components/ProductTile.js (hunk 14 above)
Comment (elric97) [inline src/Components/ProductTile.js:98]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const handleDownVote = () => {
```

File: src/Components/ProductTile.js (hunk 21 above)
Comment (elric97) [inline src/Components/ProductTile.js:9]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import DialogContent from "@mui/material/DialogContent";
```

File: src/Components/ProductTile.js (hunk 16 above)
Comment (elric97) [inline src/Components/ProductTile.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Button from "@mui/material/Button";
```

File: src/Components/ProductTile.js (hunk 27 above)
Comment (elric97) [inline src/Components/ProductTile.js:12]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import Service from "../Service";
```

File: src/Components/ProductTile.js (hunk 15 above)
Comment (elric97) [inline src/Components/ProductTile.js:37]: Codacy has a fix for the issue: [Missing space after =>.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const increaseUpVote = () => {
```

File: src/Components/ProductTile.js (hunk 29 above)
Comment (elric97) [inline src/Components/ProductTile.js:102]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
    Service.post("removeVote", form)
```

File: src/Components/ProductTile.js (hunk 31 above)
Comment (elric97) [inline src/Components/ProductTile.js:42]: Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
      .then((data) => 
```

File: src/Components/ProductTile.js (hunk 18 above)
Comment (elric97) [inline src/Components/ProductTile.js:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
import React from "react";
```

File: src/Components/ProductTile.js (hunk 22 above)
Comment (palvitgarg99) [inline src/Components/ProductTile.js]: Fixed!!

File: src/Components/ProductTile.js (hunk 30 above)
Comment (elric97) [inline src/Components/ProductTile.js:26]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/elric97/feature-hunt/pullRequest?prid=8496218)
```suggestion
  const loggedin = username !== ""?true:false;
```

**Pull request merged**